bot/internal/bot/docpaths.go

```diff
@@ -0,0 +1,122 @@
+package bot
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/shared-workflows/bot/internal/github"
+	"github.com/gravitational/trace"
+)
+
+// DocsConfig represents the structure of the config.json file found in the docs
+// directory of a Teleport repo. We only use Redirects, so other fields have the
+// "any" type.
+type DocsConfig struct {
+	Navigation any
+	Variables  any
+	Redirects  RedirectConfig
+}
+
+type RedirectConfig []Redirect
+
+type Redirect struct {
+	Source      string
+	Destination string
+	Permanent   bool
+}
+
+// CheckDocsPathsForMissingRedirects takes the relative path to a
+// gravitational/teleport clone. It assumes that there is a file called
+// "docs/config.json" at the root of the directory that lists redirects in the
+// "redirects" field.
+//
+// CheckDocsPathsForMissingRedirects checks whether a PR has renamed or deleted
+// any docs files and, if so, returns an error if any of these docs files does
+// not correspond to a new redirect in the configuration file.
```

Comment:
Looks like two separate descriptions here - did you mean to keep both paragraphs or was one meant to replace the other?